### DISCUSSION TITLE
Any clues on how to handle session ?

### DISCUSSION THREAD

**Discussion by tynods** [OxygenFramework/Oxygen.jl#163] [Q&A] [answered]
Hi,

I am wondering how to handle sessions. Do I need to use Cookies with HTTP.jl ?
Thanks

**Comment by ndortega** [comment, marked as answer]
Hi @tynods,

You're correct, you'll have to setup and manage sessions with cookies using HTTP. I'd recommend storing your session values in memory to get things working quickly, and then switch over to an actual db for persistence when it comes to deployment.

In the past I've been hesitant to add session handling to Oxygen, since the implementation can vary between apps based on their own security & performance requirements. 

The most I could do is write a wrapper around the cookies api and automatically generate and store session IDs in memory and then offer some hooks to let people customize their own persistence strategy.

**Reply by tynods** [reply]
Hi @ndortega ,

Thanks for you reply. Well I assume I can try to do it the HTTP's way. But I am not sure how to manage this with the formatting helper's functions.

Your wrapper idea and sessions id seem fine. I am sure it will help many people and add a great value to Oxygen ;-)

**Reply by tynods** [reply]
Well, I just found to to do it. I didn't realize that the helper function return an HTTP::Response. I was think that it was only the string representation. Thanks.

Your wrapping around all this is stiil a good idea :-)
